xsdata creates two classes for the same type

I am not sure what is going on, but there seems to be two dataclasses defined from the same type.  The same problem shows up in different schemas from here:  http://www.accellera.org/XMLSchema

Specifically the problematic type is portAccessType defined in:

http://www.accellera.org/XMLSchema/SPIRIT/1.5/port.xsd
http://www.accellera.org/XMLSchema/SPIRIT/1685-2009/port.xsd
http://www.accellera.org/XMLSchema/IPXACT/1685-2014/port.xsd
http://www.accellera.org/XMLSchema/IPXACT/1685-2022/port.xsd

They all show the same issue.

I see two python files (`port_access_type.py` and `port_access_type_1.py`) generated with classes `PortAccessType` and `PortAccessType1` respectively.  Both of these have:
```python
class Meta:
    name = "portAccessType"
```
Here are the two files:
***port_access_type.py***
```python
from dataclasses import dataclass, field
from typing import Optional

from org.accellera.ipxact.v1685_2014.simple_port_access_type import (
    SimplePortAccessType,
)

__NAMESPACE__ = "http://www.accellera.org/XMLSchema/IPXACT/1685-2014"


@dataclass(slots=True)
class PortAccessType:
    """Indicates how a netlister accesses a port.

    'ref' means accessed by reference (default) and 'ptr' means accessed
    by pointer.
    """

    class Meta:
        name = "portAccessType"
        namespace = "http://www.accellera.org/XMLSchema/IPXACT/1685-2014"

    value: Optional[SimplePortAccessType] = field(
        default=None,
        metadata={
            "required": True,
        },
    )
```
***port_access_type1.py***
```python
from collections.abc import Iterable
from dataclasses import dataclass, field
from typing import Optional

from org.accellera.ipxact.v1685_2014.leaf_access_handle import LeafAccessHandle
from org.accellera.ipxact.v1685_2014.port_access_type import PortAccessType

__NAMESPACE__ = "http://www.accellera.org/XMLSchema/IPXACT/1685-2014"


@dataclass(slots=True)
class PortAccessType1:
    """
    :ivar port_access_type: Indicates how a netlister accesses a port.
        'ref' means accessed by reference (default) and 'ptr' means
        accessed through a pointer.
    :ivar access_handles:
    """

    class Meta:
        name = "portAccessType"

    port_access_type: Optional[PortAccessType] = field(
        default=None,
        metadata={
            "name": "portAccessType",
            "type": "Element",
            "namespace": "http://www.accellera.org/XMLSchema/IPXACT/1685-2014",
        },
    )
    access_handles: Optional["PortAccessType1.AccessHandles"] = field(
        default=None,
        metadata={
            "name": "accessHandles",
            "type": "Element",
            "namespace": "http://www.accellera.org/XMLSchema/IPXACT/1685-2014",
        },
    )

    @dataclass(slots=True)
    class AccessHandles:
        access_handle: Iterable[LeafAccessHandle] = field(
            default_factory=list,
            metadata={
                "name": "accessHandle",
                "type": "Element",
                "namespace": "http://www.accellera.org/XMLSchema/IPXACT/1685-2014",
                "min_occurs": 1,
            },
        )
```

I see a number of duplicate files with `*1.py`, `*2.py` with the same class Meta names!

http://www.accellera.org/XMLSchema/SPIRIT/1685-2009-VE-1.0/

http://www.accellera.org/XMLSchema

Here are the commands I use to generate the binding classes.

```bash
$ xsdata generate --slots --subscriptable-types --generic-collections --relative-imports --structure-style filenames --package org.accellera.spirit.v1_5 http://www.accellera.org/XMLSchema/SPIRIT/1.5/index.xsd

$ xsdata generate --slots --subscriptable-types --generic-collections --relative-imports --structure-style filenames --package org.accellera.spirit.v1685_2009 http://www.accellera.org/XMLSchema/SPIRIT/1685-2009/index.xsd

$ xsdata generate --slots --subscriptable-types --generic-collections --relative-imports --structure-style filenames --package org.accellera.spirit.v1685_2014 http://www.accellera.org/XMLSchema/SPIRIT/1685-2014/index.xsd

$ xsdata generate --slots --subscriptable-types --generic-collections --relative-imports --structure-style filenames --package org.accellera.spirit.v1685_2022 http://www.accellera.org/XMLSchema/SPIRIT/1685-2022/index.xsd



$ xsdata generate --slots --subscriptable-types --generic-collections --relative-imports --structure-style filenames --package org.accellera.spirit.v1685_2009.ve http://www.accellera.org/XMLSchema/SPIRIT/1685-2009-VE-1.0/index.xsd

$ xsdata generate --slots --subscriptable-types --generic-collections --relative-imports --structure-style filenames --package org.accellera.spirit.v1685_2009.ve.ams http://www.accellera.org/XMLSchema/SPIRIT/1685-2009-VE-1.0/AMS/index.xsd

$ xsdata generate --slots --subscriptable-types --generic-collections --relative-imports --structure-style filenames --package org.accellera.spirit.v1685_2009.ve.core http://www.accellera.org/XMLSchema/SPIRIT/1685-2009-VE-1.0/CORE/index.xsd

$ xsdata generate --slots --subscriptable-types --generic-collections --relative-imports --structure-style filenames --package org.accellera.spirit.v1685_2009.ve.pdp http://www.accellera.org/XMLSchema/SPIRIT/1685-2009-VE-1.0/PDP/index.xsd

$ xsdata generate --slots --subscriptable-types --generic-collections --relative-imports --structure-style filenames --package org.accellera.spirit.v1685_2009.ve.power http://www.accellera.org/XMLSchema/SPIRIT/1685-2009-VE-1.0/POWER/index.xsd
```

I am not sure if the problem is with the XSDs or a bug in xsdata.

Here are some problematic xml files that fail to serialize:

[xml.tar.gz](https://github.com/user-attachments/files/18053219/xml.tar.gz)

For `scmladaptor.xml`, I get:

```
.venv/lib/python3.12/site-packages/xsdata/formats/dataclass/parsers/mixins.py:67: in from_string
    return self.from_bytes(source.encode(), clazz, ns_map)
.venv/lib/python3.12/site-packages/xsdata/formats/dataclass/parsers/mixins.py:88: in from_bytes
    return self.parse(io.BytesIO(source), clazz, ns_map)
.venv/lib/python3.12/site-packages/xsdata/formats/dataclass/parsers/bases.py:59: in parse
    result = handler.parse(source, ns_map)
.venv/lib/python3.12/site-packages/xsdata/formats/dataclass/parsers/handlers/lxml.py:41: in parse
    return self.process_context(ctx, ns_map)
.venv/lib/python3.12/site-packages/xsdata/formats/dataclass/parsers/handlers/lxml.py:59: in process_context
    self.parser.start(
.venv/lib/python3.12/site-packages/xsdata/formats/dataclass/parsers/bases.py:95: in start
    child = item.child(qname, attrs, ns_map, len(objects))

>           raise ParserError(f"Unknown property {self.meta.qname}:{qname}")
E           xsdata.exceptions.ParserError: Unknown property {http://www.accellera.org/XMLSchema/IPXACT/1685-2022}portAccessType:{http://www.accellera.org/XMLSchema/IPXACT/1685-2022}portAccessType

.venv/lib/python3.12/site-packages/xsdata/formats/dataclass/parsers/nodes/element.py:469: ParserError


E           Failed: Failed to deserialize XML file '/proj/mmip_sde_er_users07/akhailta/Work/Python/EDA/pyipxact/tests/xml/ipxact/v1685_2022/Leon2/NXP.com/adaptor_scml/scmladaptor.xml' using class '<class 'org.accellera.ipxact.v1685_2022.abstractor.Abstractor'>': Unknown property {http://www.accellera.org/XMLSchema/IPXACT/1685-2022}portAccessType:{http://www.accellera.org/XMLSchema/IPXACT/1685-2022}portAccessType
```

Somehow portAccessType is not correctly bound to the correct class.

There are many examples here:

- https://www.accellera.org/images/activities/committees/ip-xact/Leon2_1685-2014.zip
- https://www.accellera.org/images/activities/committees/ip-xact/Leon2_1685-2022.zip

Matching these schemas:

- http://www.accellera.org/XMLSchema/IPXACT/1685-2014/index.xsd
- http://www.accellera.org/XMLSchema/IPXACT/1685-2022/index.xsd

Out of 618 xml files, only 19 are failing.  I think both related to `portAccessType` of `Component` and 'Abstractor` classes/types.

```
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2014 - xml/ipxact/v1685_2014/Leon2/NXP.com/serial_device/serial_device.xml]
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2014 - xml/ipxact/v1685_2014/Leon2/NXP.com/adaptor_scml/scmladaptor.xml]
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2014 - xml/ipxact/v1685_2014/Leon2/NXP.com/uart_scml/uart_scml.xml]
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2014 - xml/ipxact/v1685_2014/Leon2/spiritconsortium.org/Leon2TLM/uart_tac/1.0/uart_tac.xml]
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2014 - xml/ipxact/v1685_2014/Leon2/spiritconsortium.org/Leon2TLM/pv2tac/1.0/pv2tac.xml]
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2014 - xml/ipxact/v1685_2014/Leon2/spiritconsortium.org/Leon2TLM/apbbus/1.4/apbbus8.xml]
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2014 - xml/ipxact/v1685_2014/Leon2/spiritconsortium.org/Leon2TLM/serial_device/1.0/serial_device.xml]
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2014 - xml/ipxact/v1685_2014/Leon2/spiritconsortium.org/Leon2TLM/scmlAdaptor/1.0/scmlAdaptor.xml]
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2014 - xml/ipxact/v1685_2014/Leon2/spiritconsortium.org/Leon2TLM/uart_scml/1.0/uart_scml.xml]
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2022 - xml/ipxact/v1685_2022/Leon2/spiritconsortium.org/Leon2TLM/uart_scml/1.0/uart_scml.xml]
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2022 - xml/ipxact/v1685_2022/Leon2/spiritconsortium.org/Leon2TLM/pv2tac/1.0/pv2tac.xml]
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2022 - xml/ipxact/v1685_2022/Leon2/spiritconsortium.org/Leon2TLM/uart_tac/1.0/uart_tac.xml]
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2022 - xml/ipxact/v1685_2022/Leon2/spiritconsortium.org/Leon2TLM/serial_device/1.0/serial_device.xml]
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2022 - xml/ipxact/v1685_2022/Leon2/spiritconsortium.org/Leon2TLM/apbbus/1.4/apbbus8.xml]
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2022 - xml/ipxact/v1685_2022/Leon2/spiritconsortium.org/Leon2TLM/scmlAdaptor/1.0/scmlAdaptor.xml]
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2022 - xml/ipxact/v1685_2022/Leon2/NXP.com/adaptor_scml/scmladaptor.xml]
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2022 - xml/ipxact/v1685_2022/Leon2/NXP.com/uart_scml/uart_scml.xml]
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2022 - xml/ipxact/v1685_2022/Leon2/NXP.com/serial_device/serial_device.xml]
FAILED tests/test_xml.py::test_parse_xml[ipxact - v1685_2022 - xml/ipxact/v1685_2022/accellera.org/Sample/SampleComponent/1.0/xml/SampleComponent.xml]
========================================================================
19 failed, 618 passed, 25 warnings in 13.32s ========================================================================
```

--- a/tests/codegen/handlers/test_process_attributes_types.py
+++ b/tests/codegen/handlers/test_process_attributes_types.py
@@ -344,25 +344,28 @@ class ProcessAttributeTypesTests(FactoryTestCase):
         complex_type = ClassFactory.create(qname="a", tag=Tag.COMPLEX_TYPE)
         simple_type = ClassFactory.create(qname="a", tag=Tag.SIMPLE_TYPE)
 
-        actual = self.processor.find_dependency(attr_type, Tag.ELEMENT)
+        actual = self.processor.find_dependency(element, attr_type, Tag.ELEMENT)
         self.assertIsNone(actual)
 
         self.processor.container.add(simple_type)
-        actual = self.processor.find_dependency(attr_type, Tag.ELEMENT)
+        actual = self.processor.find_dependency(element, attr_type, Tag.ELEMENT)
         self.assertEqual(simple_type, actual)
 
         self.processor.container.add(complex_type)
-        actual = self.processor.find_dependency(attr_type, Tag.ELEMENT)
+        actual = self.processor.find_dependency(element, attr_type, Tag.ELEMENT)
         self.assertEqual(complex_type, actual)
 
         self.processor.container.add(element)
-        actual = self.processor.find_dependency(attr_type, Tag.ELEMENT)
+        actual = self.processor.find_dependency(complex_type, attr_type, Tag.ELEMENT)
         self.assertEqual(element, actual)
 
-        actual = self.processor.find_dependency(attr_type, Tag.SIMPLE_TYPE)
+        actual = self.processor.find_dependency(element, attr_type, Tag.ELEMENT)
+        self.assertEqual(complex_type, actual)
+
+        actual = self.processor.find_dependency(element, attr_type, Tag.SIMPLE_TYPE)
         self.assertEqual(simple_type, actual)
 
-        actual = self.processor.find_dependency(attr_type, Tag.EXTENSION)
+        actual = self.processor.find_dependency(element, attr_type, Tag.EXTENSION)
         self.assertEqual(simple_type, actual)
 
     def test_update_restrictions(self):

--- a/xsdata/codegen/handlers/process_attributes_types.py
+++ b/xsdata/codegen/handlers/process_attributes_types.py
@@ -113,16 +113,19 @@ class ProcessAttributeTypes(RelativeHandlerInterface):
         if attr.restrictions.pattern:
             cls.reset_attribute_type(attr_type)
 
-    def find_dependency(self, attr_type: AttrType, tag: str) -> Optional[Class]:
+    def find_dependency(
+        self, target: Class, attr_type: AttrType, tag: str
+    ) -> Optional[Class]:
         """Find the source type from the attr type and tag.
 
         Avoid conflicts by selecting any matching type by qname and preferably:
-            1. Match the candidate object tag
-            2. Match element again complexType
+            1. Match element again complexType with no self reference
+            2. Match the candidate object tag
             3. Match non element and complexType
             4. Anything
 
         Args:
+            target: The target class instance
             attr_type: The attr type instance
             tag: The xml tag name, e.g. Element, Attribute, ComplexType
 
@@ -130,8 +133,10 @@ class ProcessAttributeTypes(RelativeHandlerInterface):
             The source class or None if no match is found
         """
         conditions = (
+            lambda obj: tag == Tag.ELEMENT
+            and obj.tag == Tag.COMPLEX_TYPE
+            and obj is not target,
             lambda obj: obj.tag == tag,
-            lambda obj: tag == Tag.ELEMENT and obj.tag == Tag.COMPLEX_TYPE,
             lambda obj: not obj.is_complex_type,
             lambda x: True,
         )
@@ -184,7 +189,7 @@ class ProcessAttributeTypes(RelativeHandlerInterface):
             attr: The attr instance
             attr_type: The attr type instance
         """
-        source = self.find_dependency(attr_type, attr.tag)
+        source = self.find_dependency(target, attr_type, attr.tag)
         if not source:
             logger.warning("Reset absent type: %s", attr_type.name)
             self.reset_attribute_type(attr_type)

